### PR TITLE
[inductor] switch CppCodeCache to new cpp_builder.

### PR DESCRIPTION
**Pull request by xuhancn**
Changes:
1. switch CppCodeCache to new cpp_builder.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/codecache.py [hunk 1]
```diff
@@ -2073,10 +2073,16 @@ def load_async(cls, source_code: str, cuda=False, submit_fn=None, extra_flags=()
...
+            output_name, output_dir = get_name_and_dir_from_output_file_path(input_path)
+            """
+            If `fb_code` env, it need to be dispatched to original `compile_file` function.
+            So, we still need to prepare parameters for the function: `input_path` and `fb_output_path`.
+            """
+            fb_output_path = input_path[:-3] + "so"
             future: Optional[Future[Any]] = None
             lib = None
-            worker_fn = functools.partial(
+            """
```
Comment (jansel) [inline torch/_inductor/codecache.py:2084]: remove old code

**Review by jansel — DISMISSED** [review]

File: torch/_inductor/codecache.py (hunk 1 above)
Comment (xuhancn) [reply torch/_inductor/codecache.py]: Done.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**